### ISSUE TITLE
Command hierarchy middleware

### ISSUE THREAD

**Issue by llonchj** [spf13/cobra#764]
I have the following command hierarchy: 

```
rootCmd (meant to initialize logging)
+--subCmd (meant to initialize a external service i.e. Redis)
   +--subSubCmd (meant to perform an action)
```

I would like the execution of subSubCmd to run a middleware pattern to setup and teardown each inheritance level.

The current PersistentPreRun and PersistentPostRun does something similar but [stops at the first command with the function](https://github.com/spf13/cobra/blob/master/command.go#L744) and also interates from parent.

Is there an existing mechanism in cobra to do that? If not, do you think it will be an interesting feature?

**Comment by llonchj** [comment]
Related to #252 and and fixed in [PR 714](https://github.com/spf13/cobra/pull/714 )

**Comment by johnSchnake** [comment]
That PR did get closed from inactivity but #252 is still open and @poy did put up a PR for it which gated the behavior.

I'm going to close this issue as duplicate as a result; please watch the other issue for a resolution/discussion.
